cgrok/grok_match.c

```diff
@@ -52,8 +52,7 @@ int grok_match_walk_next(const grok_match_t *gm,
   }
 
   *namelen = gct->name_len;
-  *name = malloc(*namelen);
-  memcpy(*name, gct->name, *namelen);
+  *name = gct->name;
```

Comment:
Is this safe to do? I assume a new string was `malloc`ed because the lifetime of the `gct` pointer wasn't long or predictable enough for the caller of `grok_match_walk_next`.